src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -421,34 +435,65 @@ private boolean isIntrospectionType(GraphQLNamedType type) {
+                        .parentType(GraphQLInputObjectType.class)
+                        .elementType(GraphQLInputObjectField.class)
+                        .build();
+                Comparator<? super GraphQLSchemaElement> comparator = options.comparatorRegistry.getComparator(environment);
 
-            out.format("input %s%s {\n", type.getName(), directivesString(GraphQLInputObjectType.class, type.getDirectives()));
-            visibility.getFieldDefinitions(type)
-                    .stream()
-                    .sorted(comparator)
-                    .forEach(fd -> {
-                        printComments(out, fd, "  ");
-                        out.format("  %s: %s",
-                                fd.getName(), typeString(fd.getType()));
-                        Object defaultValue = fd.getDefaultValue();
-                        if (defaultValue != null) {
-                            String astValue = printAst(defaultValue, fd.getType());
-                            out.format(" = %s", astValue);
-                        }
-                        out.format(directivesString(GraphQLInputObjectField.class, fd.getDirectives()));
-                        out.format("\n");
-                    });
-            out.format("}\n\n");
+                out.format("input %s%s {\n", type.getName(), directivesString(GraphQLInputObjectType.class, type.getDirectives()));
+                visibility.getFieldDefinitions(type)
+                        .stream()
+                        .sorted(comparator)
+                        .forEach(fd -> {
+                            printComments(out, fd, "  ");
+                            out.format("  %s: %s",
+                                    fd.getName(), typeString(fd.getType()));
+                            Object defaultValue = fd.getDefaultValue();
+                            if (defaultValue != null) {
+                                String astValue = printAst(defaultValue, fd.getType());
+                                out.format(" = %s", astValue);
+                            }
+                            out.format(directivesString(GraphQLInputObjectField.class, fd.getDirectives()));
+                            out.format("\n");
+                        });
+                out.format("}\n\n");
+            }
         };
     }
 
+    /**
+     * This will return true if the options say to use the AST and we have an AST element
+     *
+     * @param definition the AST type definition
+     * @return true if we should print using AST nodes
+     */
+    private boolean shouldPrintAsAst(TypeDefinition definition) {
+        return definition != null && false; // TODO - get this from options
+    }
+
+    /**
+     * This will print out a runtime graphql schema element using its contained AST type definition.  This
+     * must be guarded by a called to `shouldPrintAsAst`
```

Comment:
```suggestion
     * must be guarded by a called to {@link #shouldPrintAsAst(TypeDefinition)}
```
So you get nice click throughs and if the method name changes this should also.